src/nunavut/lang/c/support/nunavut_common.h

```diff
@@ -0,0 +1,389 @@
+/*
+ *
+ * UAVCAN common serialization support routines.                                                                     +-+ +-+
+ * 
+ * Not meant to be included directly by a user. Please include nunavut.h
+ * or nunavut_le.h instead, depending on your platform architecture.
+ * 
+ *                                                                                                            \  -  /
+ *  This software is distributed under the terms of the MIT License.                                            ---
+ *                                                                                                               o
+ * +------------------------------------------------------------------------------------------------------------------+
+ */
+#ifndef NUNAVUT_SUPPORT_NUNAVUT_COMMON_H_INCLUDED
+#define NUNAVUT_SUPPORT_NUNAVUT_COMMON_H_INCLUDED
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <string.h>
+#include <float.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef uint32_t NunavutUnsignedBitLength;
+typedef int32_t NunavutSignedBitLength;
+typedef uint32_t NunavutByteLength;
```

Comment:
Do I need to add signed/unsigned? It's only used in an unsigned context.

Reply:
Sorry bad wording, do I need to have both `NunavutSignedByteLength` and `NunavutUnsignedByteLength`?

Or do I just need to name it `NunavutUnsignedByteLength`

Only the unsigned version would be used.